### PR TITLE
btree: cursor with lineal stack structure

### PR DESCRIPTION
**Pull request by pereman2**
Removed MemPage from the code in favor of an array to encode the stack of the cursor. This is both simpler and better in terms of memory access.

O_DIRECT was removed from WAL file as it introduces alignment constraints that are too hard to follow in regular appends. Maybe in the future?

### CODE REVIEW

File: core/storage/btree.rs [hunk 1]
```diff
@@ -358,16 +367,87 @@ impl BTreeCursor {
...
     fn move_to_root(&mut self) {
-        self.page
-            .replace(Some(Rc::new(MemPage::new(None, self.root_page, 0))));
+        let mem_page = self.pager.read_page(self.root_page).unwrap();
+        self.stack.borrow_mut()[0] = Some(mem_page);
+        self.cell_indices.borrow_mut()[0] = 0;
+        *self.current_page.borrow_mut() = 0;
+    }
+
+    fn push_to_stack(&self, page: Rc<RefCell<Page>>) {
```
Comment (jussisaurio) [inline core/storage/btree.rs:376]: could we name these `stack_push()`, `stack_pop()`, `stack_peek()`?

Comment (pereman2) [reply core/storage/btree.rs:376]: maybe I will extract stack to other place, maybe not who knows
